src/test/ui/intrinsics/mem-uninitialized-future-compat.rs

```diff
@@ -0,0 +1,67 @@
+#![feature(never_type)]
+#![deny(mem_uninitialized)]
+#![allow(deprecated, invalid_value, dead_code)]
+
+use std::mem::MaybeUninit;
+
+struct UninitStruct {
+    a: u32,
+    b: char,
+}
```

Comment:
Would be nice to have tests for 1-variant enum and multi-variant enum as well.

Reply:
Done. 1-variant enums act like a struct newtype (unless they have no data, in which case they're a ZST and allowed), multi-variant enums always error (I suspect no one will be too surprised if we complain about `Result<u32, !>` and not `u32`, even if they are identical in the final layout at the moment)